File: examples/baker-example/src/main/java/webshop/simple/MakePayment.java
package webshop.simple;

import com.ing.baker.recipe.annotations.FiresEvent;
import com.ing.baker.recipe.annotations.RecipeInstanceId;
import com.ing.baker.recipe.annotations.RequiresIngredient;
import com.ing.baker.recipe.javadsl.Interaction;
import webshop.simple.ingredients.PaymentInformation;
import webshop.simple.ingredients.ReservedItems;
import webshop.simple.ingredients.ShippingAddress;

public interface MakePayment extends Interaction {

    interface MakePaymentOutcome { }

    class PaymentSuccessful implements MakePaymentOutcome { }

    class PaymentFailed implements MakePaymentOutcome { }

    @FiresEvent(oneOf = {PaymentSuccessful.class, PaymentFailed.class})
    MakePaymentOutcome apply(
            @RecipeInstanceId String recipeInstanceId,
            @RequiresIngredient("reservedItems") ReservedItems reservedItems,
            @RequiresIngredient("paymentInformation") PaymentInformation paymentInformation);
}

File: examples/bakery-interaction-examples/make-payment-and-ship-items/src/main/scala/webshop/webservice/MakePayment.scala
package webshop.webservice

import com.ing.baker.recipe.annotations.FiresEvent

import scala.concurrent.Future

case class PaymentInformation(info: String)

sealed trait MakePaymentOutput

case class PaymentSuccessful() extends MakePaymentOutput

case class PaymentFailed() extends MakePaymentOutput

trait MakePayment {

  @FiresEvent(oneOf = Array(classOf[PaymentSuccessful], classOf[PaymentFailed]))
  def apply(processId: String, items: ReservedItems, payment: PaymentInformation): Future[MakePaymentOutput]
}